notebooks/classify/signature.py

```diff
@@ -0,0 +1,249 @@
+
+        Args:
+            exactMatches (list, optional): [List of features to remove if an exact match occurs]. Defaults to [].
+            regexes (list, optional): [List of regexes to remove if the regular expression matches any features]. Defaults to [].
+            removeDuplicates (bool, optional): [Boolean that determines if duplicates should be removed]. Defaults to True.
+            isInvariant (bool, optional): [Boolean that determines if invariant columns should be removed]. Defaults to True.
+
+        Returns:
+            [pd.Dataframe]: [Pandas dataframe that has been cleaned]
+        """
+        cleanTable = self.table.applymap(lambda x: str(x) if isinstance(x, list) or isinstance(x, dict) or isinstance(x, datetime.datetime) else x)
+        
+        # Remove features that may be continuous values (i.e. time) using regular expressions and exact matches
+        
+        for feature in cleanTable:
+            
+            # Check if this feature is included in our exactMatches to remove
+            
+            if feature in exactMatches:
+                cleanTable = cleanTable.drop([feature], axis=1)
+                # print('{feature} due to being an exact match'.format(feature=feature))
+                continue
+            
+            # Check if the values in a column are all the same, if so we remove
+            
+            if isInvariant and cleanTable[feature].nunique() <= 1:
+                # print('{feature} due to being an invariant column.'.format(feature=feature))
+                cleanTable = cleanTable.drop([feature], axis=1)
+                continue
+                
+            # Else compare to regular expressions
+                
+            for regex in regexes:
+                if re.match(regex, feature):
+                    cleanTable = cleanTable.drop([feature], axis=1)
+                    # print('{feature} due to being a match with regular expression: {regex}'.format(feature=feature, regex=regex))
+                    break
+
+        if removeDuplicates:
+            # Finds empty columns to prevent them from being dropped
+            emptyCol = []
+            for column in cleanTable:
+                # Convert to numpy
+                data = cleanTable[column].to_numpy() 
+                if (data[0] == np.nan or data[0] == '') and (data[0] == data).all():
+                    emptyCol.append(column)
+                    
+            # Copy columns over to be added back after duplicates are removed
+            col = cleanTable[emptyCol]
+            
+            # Transpose the cleaned table and drop duplicate rows. Re-transpose to get back to the original table
+            cleanTable = cleanTable.T.drop_duplicates().T
+            
+            # Add empty columns back into table and reorder
+            cleanTable = pd.concat([cleanTable, col], axis=1)
+
+        return cleanTable
+
+    def binarizeData(self, table: pd.DataFrame, removeNaN: bool = True):
```

Comment:
Not sure about this method but you might want to think about making some of these "private" - i.e. prefixing with "_"
for things that are used only internally and where you don't expect users to call directly.